flake.nix

```diff
@@ -0,0 +1,99 @@
+{
+  description = "LSP-AI: An open-source language server for AI-powered functionality";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, rust-overlay }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs { 
+          inherit system overlays; 
+          config.allowUnfree = true;
+        };
+        rustVersion = pkgs.rust-bin.stable.latest.default;
+        
+        commonEnv = {
+          OPENSSL_NO_VENDOR = 1;
+          OPENSSL_DIR = "${pkgs.openssl.dev}";
+          OPENSSL_LIB_DIR = "${pkgs.openssl.out}/lib";
+          RUSTFLAGS = "-C target-cpu=native";
+        };
+
+        commonBuildInputs = with pkgs; [ 
+          openssl 
+          openssl.dev
+          zlib
+        ];
+
+        commonNativeBuildInputs = with pkgs; [ 
+          pkg-config 
+          cmake 
+          rustVersion 
+          perl
+        ];
+
+      in rec {
+        packages = rec {
+          default = pkgs.rustPlatform.buildRustPackage ({
+            pname = "lsp-ai";
+            version = "0.7.0";
+
+            src = ./.;
+
+            cargoLock = {
+              lockFile = ./Cargo.lock;
+              allowBuiltinFetchGit = true;
+            };
+
+            nativeBuildInputs = commonNativeBuildInputs;
+            buildInputs = commonBuildInputs;
+
+            buildFeatures = [ "all" ];
+
+            doCheck = false;
+
+            meta = with pkgs.lib; {
+              description = "An open-source language server that serves as a backend for AI-powered functionality";
+              homepage = "https://github.com/SilasMarvin/lsp-ai";
+              license = licenses.mit;
+              maintainers = [ /* Add maintainers here */ ];
+            };
+          } // commonEnv);
+
+          devPackage = default.overrideAttrs (oldAttrs: {
+            name = "lsp-ai-dev";
+            
+            buildPhase = ''
+              export CARGO_HOME="/tmp/.cargo"
+              export RUSTUP_HOME="/tmp/.rustup"
+              cargo build --features all
```

Comment:
Is this going to try to build it with llama.cpp support? I don't think we want that by default. The default features are what we should have built by default.

Reply:
Sounds good

Reply:
removed a bunch of unnecessary components and did some refactoring